### PR TITLE
feat: Create dialog for users to add/remove custom servers

### LINKED ISSUE #1273: Create dialog for users to add/remove custom servers.
Screenshot from 2020-11-16 22-22-41

We need a dialog that will allow users to add/remove custom servers in a table, it should open on-click a small button beside server-select field.

Refer:
- Loan accounts table in clients view page (button)
- Add charge stepper in create-savings-account (for table)

Add remove operations should be done on localstorage by settings service.

### PR DESCRIPTION
**Pull request by sankha555**
Feat: Create dialog for users to add/remove custom servers

## Description
Created a new dialog component "Custom Server Component" under Server Selector component to add custom servers to a table and display the table in the dialog box. Added servers can also be deleted from the table. Selected servers in the table are added to the list of mifosXServers in localstorage using the settings service. This will help users to add and store custom servers to run the web app.

## Related issues and discussion
Fixes #1273 

## Screenshots
New custom server dialog box
Screenshot 2020-12-26 at 4 30 06 PM

Added a new custom dummy server "https://mifos.xoxo" to the custom servers table
Screenshot 2020-12-26 at 4 30 34 PM

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/shared/server-selector/custom-server/custom-server.component.html
Comment (BLasan) [inline src/app/shared/server-selector/custom-server/custom-server.component.html:48]: Add a new line

File: src/app/shared/server-selector/custom-server/custom-server.component.scss
Comment (BLasan) [inline src/app/shared/server-selector/custom-server/custom-server.component.scss:3]: Add a new line

File: src/app/shared/server-selector/custom-server/custom-server.component.ts
Comment (BLasan) [inline src/app/shared/server-selector/custom-server/custom-server.component.ts:1]: Add the required comment above angular imports

File: src/app/shared/server-selector/custom-server/custom-server.component.ts
Comment (BLasan) [inline src/app/shared/server-selector/custom-server/custom-server.component.ts:86]: Add a new line at the EOF

**Review by BLasan — CHANGES_REQUESTED** [review]
Please make the required changes. Add a new line at each and every EOF. Looks good for me!

**Comment by sankha555** [comment]
@BLasan Made the required changes. Please review. Thanks!

File: src/app/shared/server-selector/custom-server/custom-server.component.html
Comment (BLasan) [inline src/app/shared/server-selector/custom-server/custom-server.component.html:48]: There should be a new line after 48th

**Review by BLasan — CHANGES_REQUESTED** [review]
@sankha555 Add an empty line at the end of the code. This empty line must be presented in all ts,html and scss files

**Comment by sankha555** [comment]
> @sankha555 Add an empty line at the end of the code. This empty line must be presented in all ts,html and scss files

@BLasan I think I had already added the required newlines to all the files in commit 67797f02ddb60b1f10824d8b38b2eb5564ab29e6 in response to the changes suggested by you yesterday. Please take a look at it

**Comment by BLasan** [comment]
> > @sankha555 Add an empty line at the end of the code. This empty line must be presented in all ts,html and scss files
> 
> @BLasan I think I had already added the required newlines to all the files in commit [67797f0](https://github.com/openMF/web-app/commit/67797f02ddb60b1f10824d8b38b2eb5564ab29e6) in response to the changes suggested by you yesterday. Please take a look at it

Thanks. We have to wait until Karan review and approve this PR

**Comment by sankha555** [comment]
Thanks @BLasan ! @karantakalkar kindly take a look if you find time and provide your feedback about the utility of the PR and possible merging of the PR. It would be really helpful for me in future contributions. Thanks in advance!

**Comment by karantakalkar** [comment]
LGTM, but we will need to wait till travis builds are back.

**Comment by Abhirup-99** [comment]
> LGTM, but we will need to wait till travis builds are back.

Travis builds are now migrated to one time free plan of 10k credits aka 1k Linux builds only. After this, you have to move to one of the paid plans. It would be better if you move to Github Actions.
I would be happy to push a pr, just did it for my company.

**Comment by Abhirup-99** [comment]
1 commit per pr, please squash them.

**Comment by BLasan** [comment]
> @BLasan Made the required changes. Please review. Thanks!

shall we squash the commits and rename the commit message as **feat: Create dialog for users to add/remove custom servers**

**Review by apurv-1 — APPROVED** [review]

**Pull request closed without merging**
